src/wh_client.c

```diff
@@ -438,8 +438,14 @@ int wh_Client_KeyExportResponse(whClientContext* c, uint8_t* label,
                 XMEMCPY(out, packOut, packet->keyExportRes.len);
                 *outSz = packet->keyExportRes.len;
             }
-            if (label != NULL)
-                XMEMCPY(label, packet->keyExportRes.label, labelSz);
+            if (label != NULL) {
+                if (labelSz > sizeof(packet->keyExportRes.label)) {
+                    XMEMCPY(label, packet->keyExportRes.label,
+                        WOLFHSM_NVM_LABEL_LEN);
```

Comment:
we should probably have the memcpy size argument correspond to the size check we just did

Reply:
yeah that's a little messy, still correct but it would be better to use sizeof the actual buffer